### PR TITLE
Unify message processing for api

### PR DESCRIPTION
**Pull request by OttoVT**

### CODE REVIEW

**Comment by OttoVT** [comment]
Do not merge at this point

File: src/Api.Common/Persistence/BrokerAccounts/BrokerAccountRepository.cs [hunk 1]
```diff
@@ -204,35 +222,55 @@ public async Task UpdateVaultNameAsync(long vaultId, string vaultName)
...
-                Version = default
-            });
-
-            try
+            if (brokerAccount.Version != 0)
             {
-                await context.SaveChangesAsync();
+                affectedRowsCount = await context.BrokerAccounts
+                    .Where(x => x.Id == brokerAccount.Id &&
+                                x.Version <= brokerAccount.Version)
```
Comment (KonstantinRyazantsev) [inline src/Api.Common/Persistence/BrokerAccounts/BrokerAccountRepository.cs:234]: As we've discussed it should be Sequence, not Version. Version should never leave boundaries of a service

File: src/Api.Common/Persistence/KeyKeepers/KeyKeepersRepository.cs [hunk 2]
```diff
@@ -98,20 +100,20 @@ public async Task AddOrUpdateAsync(KeyKeeper keyKeeper)
                         CreatedAt = keyKeeper.CreatedAt,
                         UpdatedAt = keyKeeper.UpdatedAt
                     });
+            }
 
-                if (affectedRowsCount == 0)
+            if (affectedRowsCount == 0)
```
Comment (KonstantinRyazantsev) [inline src/Api.Common/Persistence/KeyKeepers/KeyKeepersRepository.cs:105]: or x.UpdateAt == y.UpatedAt?

File: src/Api.Common/Persistence/KeyKeepingConfigurations/KeyKeepingConfigurationsRepository.cs [hunk 3]
```diff
@@ -45,21 +46,22 @@ public async Task AddOrUpdateAsync(KeyKeepingConfiguration keyKeepingConfigurati
                         CreatedAt = keyKeepingConfiguration.CreatedAt,
                         UpdatedAt = keyKeepingConfiguration.UpdatedAt
                     });
+            }
 
-                if (affectedRowsCount == 0)
+            if (affectedRowsCount == 0)
```
Comment (KonstantinRyazantsev) [inline src/Api.Common/Persistence/KeyKeepingConfigurations/KeyKeepingConfigurationsRepository.cs:51]: and here

**Pull request closed without merging**